### PR TITLE
Implementation of sandboxes watcher status

### PR DESCRIPTION
**Pull request by daniel-de-vera**
This is how it looks now:

```
$ signadot local connect
signadot local connect needs root privileges for:
	- updating /etc/hosts with cluster service names
	- configuring networking to direct local traffic to the cluster

signadot local connect has been started ✓
you can check its status with: signadot local status
```
```
$ signadot local status
* runtime config: cluster test, running with root-daemon
✗ Local connection not healthy!
* failed to run sandboxes watcher ("incompatible operator version, current CLI requires operator >= 0.15.0")
```

In a successful case:

```
$ signadot local status
* runtime config: cluster test, running with root-daemon
✓ Local connection healthy!
    * port-forward listening at ":46727"
    * localnet has been configured
    * 41 hosts accessible via /etc/hosts
    * sandboxes watcher is running                    <-------
* Connected Sandboxes:
    - daniel-hotrod-route
        * Routing Key: b7juzn2hsnl1j
        - local-route: routing from Deployment/route in namespace "hotrod"
            - remote port 8083 -> localhost:8083
        ✓ connection ready

```

### CODE REVIEW

File: internal/command/local/printers.go [hunk 1]
```diff
@@ -238,6 +240,42 @@ func getRawPortforward(cfg *config.LocalStatus, ciConfig *config.ConnectInvocati
 	return result
 }
 
+func getRawWatcher(cfg *config.LocalStatus, watcher *commonapi.WatcherStatus,
+	statusMap map[string]any) any {
+	var result any
+
+	if cfg.Details {
+		// Details view
+		result = statusMap["watcher"]
```
Comment (scott-cotton) [inline internal/command/local/printers.go:249]: can we just return `statusMap["watcher"]` here?

File: internal/command/local/printers.go [hunk 2]
```diff
@@ -238,6 +240,42 @@ func getRawPortforward(cfg *config.LocalStatus, ciConfig *config.ConnectInvocati
...
+		type PrintableWatcher struct {
+			Healthy         bool   `json:"healthy"`
+			LastErrorReason string `json:"lastErrorReason,omitempty"`
+		}
+
+		result = &PrintableWatcher{
+			Healthy: false,
+		}
+
+		if watcher != nil {
```
Comment (scott-cotton) [inline internal/command/local/printers.go:261]: Would suggest `if watcher == nil || watcher.Health == nil { return watcher }`

File: internal/command/local/printers.go (hunk 1 above)
Comment (daniel-de-vera) [reply internal/command/local/printers.go]: It's done, I took the same approach in the other related functions.

File: internal/command/local/printers.go (hunk 2 above)
Comment (daniel-de-vera) [reply internal/command/local/printers.go]: Same as above.

**Comment by scott-cotton** [comment]
I would suggest adding the local machine id to the status, as otherwise it is unclear what set of sandboxes are reconnected and why.

**Comment by scott-cotton** [comment]
I find it odd that `signadot local connect --wait` finishes successfully after showing:

```
signadot local connect has been started ✓
* runtime config: cluster minikube, running with root-daemon
✓ Local connection healthy!
    * port-forward listening at ":56487"
    * localnet has been configured
    * 20 hosts accessible via /etc/hosts
    * sandboxes watcher is running
* Connected Sandboxes:
    - my-local-1
        * Routing Key: pw9pl10kzw6yg
        - local-route: routing from Deployment/route in namespace "hotrod"
            - remote port 8083 -> localhost:8083
        ✗ connection not ready
    - my-local
        * Routing Key: zg9xg98uv17kd
        - local-route: routing from Deployment/route in namespace "hotrod"
            - remote port 8083 -> localhost:8083
        ✗ connection not ready
```

I'm not sure what to do about this, as technically the local connection could be ready while the sandboxes not, but given that `signadot local connect` now _reconnects_ sandboxes, it is strange that `--wait` doesn't wait for those reconnected sandboxes to be ready.

wdyt?

**Comment by daniel-de-vera** [comment]
> I would suggest adding the local machine id to the status, as otherwise it is unclear what set of sandboxes are reconnected and why.

I added this in the details view:

```
$ signadot local status --details
* runtime config: cluster test, running with root-daemon (config-dir: /home/ddv/.signadot, machine-id 959857d00db18b5d58363aab63e6dd6b59423b053d1e6246f2a5e80beefac2f)
✗ Local connection not healthy!
* failed to run sandboxes watcher
```

**Comment by scott-cotton** [comment]
> > I would suggest adding the local machine id to the status, as otherwise it is unclear what set of sandboxes are reconnected and why.
> 
> I added this in the details view:
> 
> ```
> $ signadot local status --details
> * runtime config: cluster test, running with root-daemon (config-dir: /home/ddv/.signadot, machine-id 959857d00db18b5d58363aab63e6dd6b59423b053d1e6246f2a5e80beefac2f)
> ✗ Local connection not healthy!
> * failed to run sandboxes watcher
> ```

That's better, but I'm not sure `--details` is visible/used enough to avoid leaving the user too mystified.
@foxish any thoughts on visibility of machineID on your end?

**Comment by scott-cotton** [comment]
> @scott-cotton, I will merge this into the parent branch, that way I can work on the backwards compatibility task on top of this (and I don't keep adding new branches and PRs, we already have several). We can address this comment after I'm done with the upcoming changes.

Can we track that somewhere?

**Review by scott-cotton — APPROVED** [review]
LGTM

**Comment by foxish** [comment]
> > > I would suggest adding the local machine id to the status, as otherwise it is unclear what set of sandboxes are reconnected and why.
> > 
> > 
> > I added this in the details view:
> > ```
> > $ signadot local status --details
> > * runtime config: cluster test, running with root-daemon (config-dir: /home/ddv/.signadot, machine-id 959857d00db18b5d58363aab63e6dd6b59423b053d1e6246f2a5e80beefac2f)
> > ✗ Local connection not healthy!
> > * failed to run sandboxes watcher
> > ```
> 
> That's better, but I'm not sure `--details` is visible/used enough to avoid leaving the user too mystified. @foxish any thoughts on visibility of machineID on your end?

This seems reasonable to me - let's do this for now and then if we need more visibility in future, we'll add it in more places.

**Comment by daniel-de-vera** [comment]
> I find it odd that `signadot local connect --wait` finishes successfully after showing:
> 
> ```
> signadot local connect has been started ✓
> * runtime config: cluster minikube, running with root-daemon
> ✓ Local connection healthy!
>     * port-forward listening at ":56487"
>     * localnet has been configured
>     * 20 hosts accessible via /etc/hosts
>     * sandboxes watcher is running
> * Connected Sandboxes:
>     - my-local-1
>         * Routing Key: pw9pl10kzw6yg
>         - local-route: routing from Deployment/route in namespace "hotrod"
>             - remote port 8083 -> localhost:8083
>         ✗ connection not ready
>     - my-local
>         * Routing Key: zg9xg98uv17kd
>         - local-route: routing from Deployment/route in namespace "hotrod"
>             - remote port 8083 -> localhost:8083
>         ✗ connection not ready
> ```
> 
> I'm not sure what to do about this, as technically the local connection could be ready while the sandboxes not, but given that `signadot local connect` now _reconnects_ sandboxes, it is strange that `--wait` doesn't wait for those reconnected sandboxes to be ready.
> 
> wdyt?

@scott-cotton, I implemented this in PR: https://github.com/signadot/cli/pull/95

**Pull request merged**